agent/php_agent.c

```diff
@@ -87,6 +90,32 @@ zval* nr_php_get_zval_object_property(zval* object,
   return 0;
 }
 
+zval* nr_php_get_zval_base_exception_property(zval* exception,
+                                              const char* cname) {
+  char* name;
+  zend_class_entry* ce;
+
+  if ((0 == exception) || (0 == cname) || (0 == cname[0])) {
+    return 0;
+  }
+
+  name = (char*)nr_alloca(nr_strlen(cname) + 1);
+  nr_strcpy(name, cname);
+
+  if (nr_php_is_zval_valid_object(exception)) {
+    if (nr_php_error_zval_is_exception(exception)) {
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO
+      ce = zend_get_exception_base(Z_OBJ_P(exception));
```

Comment:
This is inline with what the php source code did when we called via `nr_php_call(getTrace)`, etc. This is needed to maintain consistent results within our tests. Without getting the base class entry, certain values are incorrect for either errors/exceptions.

Reply:
Can you add the above context in a code comment?

Reply:
added in [18f7088](https://github.com/newrelic/newrelic-php-agent/pull/825/commits/18f708807d20e7056fcaa7d1f80a14a8a985878a)